### ISSUE TITLE
Release version 3.1

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#681]
Given the public release of WikiMedia ZIM files with WebP image compression as standard, it is important to get a new release out as soon as practicable. Since Travis tests are currently broken, we can only release code that has already passed tests and has been merged into master for now, so I suggest we do not wait on any of the PRs currently in the pipeline. If that is agreed, then below are the usual steps:

* [x] Update the changelog file
* [x] Further tests 
* [x] Move open issues to 3.2
* [x] Create a tag on git:
* [x] Generate unsigned extensions
* [x] Upload the unsigned extensions to Stores:
  - [x] Firefox
  - [x] Chrome
  - [x] Edge - https://partner.microsoft.com/en-us/dashboard/microsoftedge/
  - [x] Ubuntu touch - https://open-store.io
* [x] Prepare the source code for version 3.2
* [x] When available, copy the signed extension binaries to https://download.kiwix.org/release/browsers/ (new location)
* [x] Update the kiwix.github.io implementation
* [x] Publicize these new versions

**Comment by mossroy** [comment]
I agree.
We'll have to do that without Travis. Never mind : we don't need it to release

**Comment by Jaifroid** [comment]
I forgot to add a release date to the Changelog before merging, so I'll update that with a separate commit once further tests have been completed and before creating the tag.

**Comment by Jaifroid** [comment]
@ernesst We are soon going to release v3.1 with WebP image support. We have an issue in that Travis CI is no longer working (due to a change in their policy), so the last auto-generated .click file we have is:

https://download.kiwix.org/nightly/2020-12-03/kiwix-ubuntu-touch-3.1commit-3457998.click 

Fortunately this contains all the changes we are planning to release (and we'll generate a new file manually for the actual release), if you are able to test that one with a ZIM containing WebP images, it would be very helpful. This Ray Charles is a suitable ZIM for testing

http://download.kiwix.org/zim/wikipedia/wikipedia_en_ray_charles_maxi_2020-12.zim 

(ignore the missing H1 article heading, this is a scraping error I've reported in https://github.com/openzim/mwoffliner/issues/1322).

Many thanks in advance.

**Comment by Jaifroid** [comment]
@mossroy @ernesst Are there any more manual tests pending, or can I go ahead and create the release tag?

**Comment by mossroy** [comment]
I'll do some tests on my Firefox OS device today

**Comment by ernesst** [comment]
All,

I just test this .click with the zim. I didn't find any regression.

Ernesst.

Le Sun, 6 Dec 2020 10:17:04 -0800, Jaifroid a écrit :
> @ernesst We are soon going to release v3.1 with WebP image 
> support. We have an issue in that Travis CI is no longer working 
> (due to a change in their policy), so the last auto-generated 
> .click file we have is:
>
> https://download.kiwix.org/nightly/2020-12-03/kiwix-ubuntu-touch-3.1commit-3457998.click 
> 
>
> Fortunately this contains all the changes we are planning to 
> release (and we'll generate a new file manually for the actual 
> release), if you are able to test that one with a ZIM containing 
> WebP images, it would be very helpful. This Ray Charles is a 
> suitable ZIM for testing
>
> http://download.kiwix.org/zim/wikipedia/wikipedia_en_ray_charles_maxi_2020-12.zim 
> 
>
> (ignore the missing H1 article heading, this is a scraping 
> error I've reported in 
> https://github.com/openzim/mwoffliner/issues/1322).
>
> Many thanks in advance.
>

-- 
Sent using Dekko from my Ubuntu device

**Comment by Jaifroid** [comment]
Great, thanks @ernesst!

**Comment by mossroy** [comment]
I tested many different ZIM files (including the last webp one you suggested) on my Firefox OS device and found no regression.
The webp polyfill works perfectly : it's really cool to be able to display them on old platforms.
And the recent improvements (in particular the sequential extracting of images) make a big difference on the impression of speed of the app on this kind of device.

I also quickly tested on Firefox 83 (as an extension, jQuery mode) and Chromium 87 (as an extension, SW mode) : both work fine with the webp ZIM file and a few other ones

So, it's a "go" on my side

**Comment by Jaifroid** [comment]
Thanks @mossroy, that's helpful.

**Comment by Jaifroid** [comment]
So we finally have the released packages for the various stores. They are:

* [kiwix-ubuntu-touch-3.1.0.click](https://download.kiwix.org/nightly/2020-12-12/kiwix-ubuntu-touch-3.1.0.click)
* [kiwix-firefoxos-3.1.0.zip](https://download.kiwix.org/nightly/2020-12-12/kiwix-firefoxos-3.1.0.zip)
* [kiwix-firefox-unsigned-listed-extension-3.1.0.zip](https://download.kiwix.org/nightly/2020-12-12/kiwix-firefox-unsigned-listed-extension-3.1.0.zip)
* [kiwix-chrome-unsigned-extension-3.1.0.zip](https://download.kiwix.org/nightly/2020-12-12/kiwix-chrome-unsigned-extension-3.1.0.zip)

I'll deal with the Edge Store. @kelson42 can you kindly deal with the Chrome store and with communication about this release, as you usually do?

**Comment by mossroy** [comment]
Version 3.1.0 is deployed on Firefox and Ubuntu Touch

**Comment by Jaifroid** [comment]
@kelson42 Could you kindly deploy the Chrome extension to Chrome store?

**Comment by kelson42** [comment]
@Jaifroid I will, sorry for the delay

**Comment by kelson42** [comment]
@Jaifroid I have published 3.1.0 on chrome store and will put the binaries on download.kiwix.org. Would you be able please to remember me how to get the zip for edge please?

**Comment by Jaifroid** [comment]
@kelson42 Thank you! The zip for Edge is exactly the same one as for Chrome.

**Comment by kelson42** [comment]
Done, will make the usual communication later today.
